resolwe/observers/models.py

```diff
@@ -0,0 +1,82 @@
+"""The model Observer model."""
+import random
+
+from django.conf import settings
+from django.db import models
+from django.db.models import Count, Q
+
+from .protocol import CHANGE_TYPE_CREATE, CHANGE_TYPE_DELETE, CHANGE_TYPE_UPDATE
+
+
+def get_random_hash():
+    """Generate a random 256-bit number as a hex string."""
+    return hex(random.randint(0, 2**128))[2:]
+
+
+class Observer(models.Model):
+    """State of a model observer."""
+
+    CHANGE_TYPES = (
+        (CHANGE_TYPE_CREATE, "create"),
+        (CHANGE_TYPE_UPDATE, "update"),
+        (CHANGE_TYPE_DELETE, "delete"),
+    )
+
+    # Table of the observed resource
+    table = models.CharField(max_length=100)
```

Comment:
Reference table by content type instead of the name as this is a more common approach in Django. Check how [GenericForeignKey](https://docs.djangoproject.com/en/4.0/ref/contrib/contenttypes/#django.contrib.contenttypes.fields.GenericForeignKey) does it.